#!/bin/sh /usr/share/dpatch/dpatch-run
## 03-gcc-call.dpatch by Fabian Greffrath <fabian@debian-unofficial.org>
## and Mike Frysinger <vapier@gentoo.org>.
##
## DP: Call C-Compiler via $(CC), consider CFLAGS, remove -static flag,
## DP: drop linking against libncurses and change include dir order.
## DP: Furthermore use standard env vars and remove pointless -I paths.

@DPATCH@
diff -urNad unace-nonfree-2.5~/makefile unace-nonfree-2.5/makefile
--- unace-nonfree-2.5~/makefile	2008-04-09 11:28:30.000000000 +0200
+++ unace-nonfree-2.5/makefile	2008-04-09 11:28:30.000000000 +0200
@@ -14,17 +14,16 @@
 SRCSsl          = $(BASIC_DIR)source
 SRCS            = $(SRCSsl)/
 
+ifdef SYSINC
 INCLSYSDIR 	= //usr/include/sys
 INCLDIR 	= //usr/include
-
-DEFINES		= -D__LINUX__ -D__unix__ -D__GCC__ -U__HAS_IO_H__
-
-ifdef DEBUG
-LSWITCHES	=
+INCLS		= -I$(INCLDIR) -idirafter $(INCLSYSDIR)
 else
-LSWITCHES	= -Wl,-s
+INCLS		=
 endif
 
+DEFINES		= -D__LINUX__ -D__unix__ -D__GCC__ -U__HAS_IO_H__
+
 ##############################################################################
 all: unace
 
@@ -47,7 +46,7 @@
   $(APPS_UNACEEXE_CFILES)
 
 unace: $(UNACEEXELIN_CFILES)
-	gcc $(LSWITCHES) -Wl,-lncurses -static -I$(INCLDIR) -I$(INCLSYSDIR) -I$(SRCSsl) $(DEFINES) $(UNACEEXELIN_CFILES) -ggdb -o$(EXECS_DIR)unace
+	$(CC) $(CFLAGS) $(LDFLAGS) $(INCLS) -I$(SRCSsl) $(DEFINES) $(UNACEEXELIN_CFILES) -ggdb -o$(EXECS_DIR)unace
 ifndef DEBUG
 	tar cfvz linunace25.tgz unace file_id.diz licence
 	#sh linpack.sh
